tests/lib/rules/no-jquery.js

```diff
@@ -169,6 +196,11 @@ eslintTester.run('no-jquery', rule, {
       errors: [
         {
           message: ERROR_MESSAGE,
+          line: 2,
+        },
+        {
+          message: ERROR_MESSAGE,
+          line: 5,
```

Comment:
The rule now reports two errors when destructuring from `Ember`

- one for the `const { $ } = Ember;` line,
- and one for the `$(body).addClass('active')` line

I think this is because eslint treats both the `key` and `value` of the `$` property as two seperate references, just with the same name.

However, this does mean the rule can now track when the `value` is renamed eg;

```js
const { $: jq } = Ember;

jq(body).addClass('active');
```

Reply:
I would prefer if `const { $ } = Ember;` was not flagged since that ends up being a duplicate violation. Using the `ReferenceTracker` approach I mentioned in the other comments, only function calls will be flagged.

Reply:
Agree on the duplicate violation point 👍🏻  I'll make sure to avoid that..

Reply:
> only function calls will be flagged

Using `ReferenceTracker` with `ReferenceTracker.CALL` means things like `$.post()` won't be marked as a violation, only `$()`.

For the following..

```js
const { $: jq } = Ember;
jq(body).addClass('active');
jq.post();
```

..which lines would you expect to be reported by `eslint`?

Reply:
I would expect `jq(body)` and `jq.post()` to be flagged.

Reply:
We might want to add a list of all the jQuery functions like `post` / `extend` / etc to the ReferenceTracker so it can find them for us.